include/aspect/simulator_access.h

```diff
@@ -937,20 +937,36 @@ namespace aspect
       n_particle_worlds() const;
 
       /**
-       * Returns a const reference to the particle world, in case anyone
+       * Returns a const reference to the vector containing all particle worlds,
+       * in case anyone wants to query something about particles.
+       */
+      const std::vector<std::unique_ptr<Particle::World<dim>>> &
+      get_particle_worlds() const;
+
+      /**
+       * Returns a reference to the vector containing all particle world,
+       * in case anyone wants to change something within the particle world.
+       * Use with care, usually you want to only let the functions within the
+       * particle subsystem change member variables of the particle world.
+       */
+      std::vector<std::unique_ptr<Particle::World<dim>>> &
+      get_particle_worlds();
+
+      /**
+       * Returns a const reference to a single particle world, in case anyone
        * wants to query something about particles.
        */
       const Particle::World<dim> &
-      get_particle_world() const;
+      get_particle_world(const unsigned int particle_world_index) const;
 
       /**
-       * Returns a reference to the particle world, in case anyone wants to
+       * Returns a reference to a single particle world, in case anyone wants to
        * change something within the particle world. Use with care, usually
        * you want to only let the functions within the particle subsystem
        * change member variables of the particle world.
        */
       Particle::World<dim> &
-      get_particle_world();
+      get_particle_world(const unsigned int particle_world_index);
```

Comment:
oh, I see we have a non-const version of this too. Anyway, the use of this function should not be encouraged.